velocidi-sdk/src/main/kotlin/com/velocidi/Util.kt

```diff
@@ -0,0 +1,88 @@
+package com.velocidi
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import java.net.URI
+import java.net.URISyntaxException
+import java.net.URL
+
+data class ApplicationInfo(
+    val appName: String,
+    val appVersion: String,
+    val androidSDK: String,
+    val device: String
+)
+
+object Util {
+    /**
+     * Obtains the necessary information to build the User-Agent(UA)
+     *
+     * @param context Android application context
+     * @return ApplicationInfo with all the relevant information to build the UA
+     */
+    fun getApplicationInfo(context: Context): ApplicationInfo {
+        val sdkVersion = Build.VERSION.SDK_INT.toString()
+        val device = if (Build.MANUFACTURER != null && Build.MODEL != null) {
+            Build.MANUFACTURER + " " + Build.MODEL
+        } else {
+            Build.DEVICE ?: ""
+        }
+
+        return try {
+            val packageManager = context.packageManager
+            val packageName = context.packageName
+
+            val applicationInfo = packageManager.getApplicationInfo(packageName, 0)
+            val packageInfo = packageManager.getPackageInfo(packageName, 0)
+
+            val appName = packageManager.getApplicationLabel(applicationInfo) ?: packageName
+            val appVersion = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                packageInfo.longVersionCode.toString()
+            } else {
+                packageInfo.versionName
+            }
+
+            ApplicationInfo(appName.toString(), appVersion.toString(), sdkVersion, device)
+        } catch (e: PackageManager.NameNotFoundException) {
+            ApplicationInfo("Unknown app", "Unknown version", sdkVersion, device)
+        }
+    }
+
+    fun buildUserAgent(appInfo: ApplicationInfo): String =
+        "${appInfo.appName}/${appInfo.appVersion} ${appInfo.androidSDK} ${appInfo.device}"
```

Comment:
The User-Agent string should follow the format similar to the defined in the document @beatriz made when exploring the SDK development, where she mentioned using this example [here](https://github.com/wikimedia/apps-android-wikipedia/blob/53fe728a995f7442a3942108b0fa3d978bed15d4/app/src/main/java/org/wikipedia/WikipediaApp.java#L211) 

It should also include the name of our SDK and the version or build number.

In the end, it should be something like `$appName/$appVersion $sdkName/$sdkVersion (Android ${Build.VERSION.RELEASE}; $deviceType}`.

We could also improve on this and prepend the `$appName/$appVersion $sdkName/$sdkVersion` to Android's default User-Agent (fetch it using ` System.getProperty("http.agent")`), which I prefer.

Reply:
The default `http.agent` comes with a lot of unnecessary information. Here's an example: 
`Dalvik/2.1.0 (Linux; U; Android 8.1.0; Android SDK built for x86 Build/OSM1.180201.007)`

Do you still think we should use this in our custom UA?

Reply:
Changed in 55c3266